src/main/java/org/opensrp/referral/controller/ReferralLocationController.java

```diff
@@ -2,15 +2,19 @@
 
 import com.google.gson.Gson;
 import org.json.JSONException;
+import org.json.JSONObject;
 import org.opensrp.referral.service.ReferralLocationService;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Controller;
 import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.ResponseBody;
 
+import static org.springframework.web.bind.annotation.RequestMethod.POST;
+
 @Controller
 @RequestMapping("/location/")
 public class ReferralLocationController {
```

Comment:
I think we will need to rename this class since it's use case is not for referral work alone.

Reply:
I suggest `CommonLocationController`